### PR TITLE
[BOSH] Add Bosh spec

### PR DESCRIPTION
**Pull request by hjaffan**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
This introduces the BOSH cli to the mix of commands that are known to autocomplete 
**What is the current behavior? (You can also link to an open issue here)**

**What is the new behavior (if this is a feature change)?**
BOSH CLI auto complete
**Additional info:**

### CODE REVIEW

**Review by cstrnt — CHANGES_REQUESTED** [review]
Hey there, thank you very much for contributing :) Just a little side-note: the `name` prop should **never** end with an `=`, because Fig takes care of this internally

File: dev/bosh.ts [hunk 1]
```diff
@@ -0,0 +1,445 @@
+import { descriptionRule } from "danger/distribution/commands/init/default-dangerfile";
```
[the remark this answers is not in the document]
Comment (hjaffan) [reply dev/bosh.ts]: Removed it in the Updated Commit

File: dev/bosh.ts [hunk 2]
```diff
@@ -0,0 +1,445 @@
...
+
+var deployments = {
```
[the remark this answers is not in the document]
Comment (hjaffan) [reply dev/bosh.ts]: Resolved in updated PR

File: dev/bosh.ts [hunk 3]
```diff
@@ -0,0 +1,445 @@
...
+    {
+      name: ["--deployment="],
+      description: "Deployment name [$BOSH_DEPLOYMENT]",
+      args: {
+        generators: deployments,
+      },
+    },
+    {
+      name: ["--column="],
+      description: "Filter to show only given column(s)",
```
[the remark this answers is not in the document]
Comment (hjaffan) [reply dev/bosh.ts:424]: resovled

File: dev/bosh.ts [hunk 4]
```diff
@@ -0,0 +1,445 @@
...
+      description: "Create or update BOSH environment",
+    },
+  ],
+  options: [
+    {
+      name: ["--version"],
+      description: "Show CLI version",
+    },
+    {
+      name: ["--config="],
```
[the remark this answers is not in the document]
Comment (hjaffan) [reply dev/bosh.ts]: It takes a file path currently.

File: dev/bosh.ts [hunk 5]
```diff
@@ -0,0 +1,443 @@
...
+      description: "Show current CPI config",
+    },
+    {
+      name: "create-env",
+      description: "Create or update BOSH environment",
+    },
+  ],
+  options: [
+    {
+      name: ["--version"],
```
Comment (QuiiBz) [inline dev/bosh.ts:383]: You can set a plain string instead of an array since there is only one value. Same for each underneath.

**Review by QuiiBz — COMMENTED** [review]
Thanks for your contribution. Could you run `npm run build` to generate the needed JS spec file, and push it?

File: dev/bosh.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (cstrnt) [reply dev/bosh.ts]: That's good! This means adding an empty arg object in Fig if you are lazy or adding `args: { template: "filePaths" }`which will make Fig display suggestions for files

**Comment by cstrnt** [comment]
Closing this in favor of #218

**Pull request closed without merging**